### PR TITLE
Change debug frame messages to adhere to JSON format

### LINKED ISSUE #8946: Debugger frames need to be in proper JSON format to provide better formatting in composer UI
**Description:**
Currently ballerina core debugger outputs frames as strings like this.
```
{
"packageName":"ballerina.http",
"frameName":"get",
"fileName":"http_caching_client.bal",
"lineID":370,"variables":[
{"scope":"Local","name":"self","type":"ballerina.http:HttpCachingClient","value":"struct HttpCachingClient {serviceUri:\\\"http:\\/\\/www.simpsonquotes.xyz\\\", config:{url:\\\"http:\\/\\/www.simpsonquotes.xyz\\\", circuitBreaker:null, timeoutMillis:60000, keepAlive:AUTO, chunking:AUTO, httpVersion:\\\"1.1\\\", forwarded:\\\"disable\\\", followRedirects:null, retryConfig:null, proxy:null, connectionThrottling:null, secureSocket:null, cache:{enabled:true, isShared:false, expiryTimeMillis:86400, capacity:8388608, evictionFactor:0.2, policy:CACHE_CONTROL_AND_VALIDATORS}, compression:AUTO, auth:null}, httpClient:{serviceUri:\\\"http:\\/\\/www.simpsonquotes.xyz\\\", config:null}, cache:{}, cacheConfig:{enabled:true, isShared:false, expiryTimeMillis:86400, capacity:8388608, evictionFactor:0.2, policy:CACHE_CONTROL_AND_VALIDATORS}}"},{"scope":"Local","name":"path","type":"string","value":"\\/quote"},{"scope":"Local","name":"request","type":"null","value":"null"},{"scope":"Local","name":"req","type":"null","value":"null"}
]}
```
suggested format

```js
{
    "packageName": "ballerina.http",
    "frameName": "get",
    "fileName": "http_caching_client.bal",
    "lineID": 370,
    "variables": [
        {
            "scope": "Local",
            "name": "self",
            "type": "ballerina.http:HttpCachingClient",
            "value": {
                "serviceUri": "http://www.simpsonquotes.xyz",
                "config": {
                    "url": "http://www.simpsonquotes.xyz",
                    "circuitBreaker": null,
                    "timeoutMillis": 60000,
                    "keepAlive": "AUTO",
                    "chunking": "AUTO",
                    "httpVersion": "1.1",
                    "forwarded": "disable",
                    "followRedirects": null,
                    "retryConfig": null,
                    "proxy": null,
                    "connectionThrottling": null,
                    "secureSocket": null,
                    "cache": {
                        "enabled": true,
                        "isShared": false,
                        "expiryTimeMillis": 86400,
                        "capacity": 8388608,
                        "evictionFactor": 0.2,
                        "policy": "CACHE_CONTROL_AND_VALIDATORS",
                        "compression": "AUTO",
                        "auth": null
                    },
                    "httpClient": {
                        "serviceUri": "http://www.simpsonquotes.xyz",
                        "config": null
                    },
                    "cacheConfig": {
                        "enabled": true,
                        "isShared": false,
                        "expiryTimeMillis": 86400,
                        "capacity": 8388608,
                        "evictionFactor": 0.2,
                        "policy": "CACHE_CONTROL_AND_VALIDATORS"
                    }
                }
            }
        },
        {
            "scope": "Local",
            "name": "path",
            "type": "string",
            "value": "/quote"
        },
        {
            "scope": "Local",
            "name": "request",
            "type": "null",
            "value": "null"
        },
        {
            "scope": "Local",
            "name": "req",
            "type": "null",
            "value": "null"
        }
    ]
}
}
```

### PR DESCRIPTION
**Pull request by pranavan15**
## Purpose
This PR fixes #8946. With this PR, debug frame messages will be changed such that they are JSON parsable. Debugger clients are expected to format and parse these frame messages.

### CODE REVIEW

**Comment by pahans** [comment]
Hi @pranavan15 @lankavitharana ,
I am currently reviewing this.Since we allow multiple versions of ballerina to use with idea/vscode plugins we might need to ensure backward compatibility as well. 
BTW, should we send this fix with the patch release? IMO this PR should be merged to master

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/dto/VariableDTO.java [hunk 1]
```diff
@@ -79,22 +78,10 @@ private String getStringValue(BValue bValue) {
...
-            BMap bmap = (BMap) bValue;
-            bValueString = "Map[" + bmap.size() + "] ";
-            bValueString = bValueString + bmap.stringValue();
-        } else if (bValue.getType().getTag() == TypeTags.RECORD) {
-            bValueString = "Record " + bValue.getType().getName() + " ";
-            bValueString = bValueString + bValue.stringValue();
         } else if (bValue.getType().getTag() == TypeTags.OBJECT) {
-            bValueString = "Object " + bValue.getType().getName() + " ";
-            bValueString = bValueString + ((BMap) bValue).absoluteStringValue();
+            bValueString = ((BMap) bValue).absoluteStringValue();
```
Comment (pahans) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/dto/VariableDTO.java]: I checked the output of "absoluteStringValue" it isnt a valid json value. Keys needs to be surrounded by quotes. 
```js
{
	config: {
		host: "0.0.0.0",
		keepAlive: "AUTO",
		secureSocket: null,
		httpVersion: "1.1",
		requestLimits: null,
		filters: [],
		timeoutMillis: 120000,
		maxPipelinedRequests: 10,
		authProviders: null,
		positiveAuthzCache: {
			enabled: true,
			capacity: 100,
			expiryTimeMillis: 5000,
			evictionFactor: 1.0
		},
		negativeAuthzCache: {
			enabled: true,
			capacity: 100,
			expiryTimeMillis: 5000,
			evictionFactor: 1.0
		}
	},
	filterContext: {
		serviceRef: {},
		serviceName: "myService1$$service$0",
		resourceName: "foo",
		attributes: {}
	},
	remoteAddress: {
		host: "localhost",
		port: 44740
	},
	localAddress: {
		host: "localhost",
		port: 9090
	},
	protocol: "http"
}
```

**Comment by pranavan15** [comment]
> hi @pranavan15 added a comment.
> Addition to that we might encounter cross platform issues in this approach.
> IMO we should use gson library for json generation to avoid issues like this.(its already dependency in ballerina parent).
> PS: gson library is going to be replaced by [JsonSerializer](https://github.com/ballerina-platform/ballerina-lang/pull/10201). Shall we try to use JsonSerializer for json generation?

@pahans +1. The JSON generation logic we have for debugger is pretty old and error proned as you pointed out. Will try to adopt JsonSerializer for the JSON generation.

**Comment by pranavan15** [comment]
Closing this PR for now. Will resend it to master branch once the existing JSON generation logic is replaced with the JsonSerializer.

**Pull request closed without merging**